android/app/src/main/java/com/microsoft/codepush/common/connection/BaseJob.java

```diff
@@ -0,0 +1,42 @@
+package com.microsoft.codepush.common.connection;
+
+import android.os.AsyncTask;
+
+import com.microsoft.codepush.common.utils.CodePushUtils;
+import com.microsoft.codepush.common.utils.FileUtils;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * This class represent basic code push job.
+ *
+ * @param <T> type of the returned parameter.
+ */
+public abstract class BaseJob<T> extends AsyncTask<Void, Void, T> {
```

Comment:
`BaseJob` name looks too generic, what about `BaseHtttJob` or so?

Reply:
Can be